### PR TITLE
Introducing custom exceptions for application policy

### PR DESCRIPTION
**Pull request by pubudu538**
ApplicationPolicyAlreadyExistsException is added.
UI gives proper error message when adding an already exist application policy.

### CODE REVIEW

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java [hunk 1]
```diff
@@ -817,12 +817,18 @@ public Response addApplicationPolicy(
             return Response.created(url).entity(new ResponseMessageBean(ResponseMessageBean.SUCCESS,
                     String.format("Application policy added successfully: [application-policy] %s",
                             applicationPolicy.getId()))).build();
-        } catch (RestAPIException e) {
-            throw e;
         } catch (AutoscalerServiceInvalidApplicationPolicyExceptionException e) {
             return Response.status(Response.Status.BAD_REQUEST).entity(new ResponseMessageBean(
                     ResponseMessageBean.ERROR, "Invalid application policy")).build();
+        } catch(AutoscalerServiceApplicationPolicyAlreadyExistsExceptionException e){
+            return Response.status(Response.Status.CONFLICT).entity(new ResponseMessageBean(
```
Comment (lahirus) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java]: You can improve the formatting in your diff. There should be a space between ')' and '{'

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java [hunk 2]
```diff
@@ -699,7 +699,7 @@ public static void addAutoscalingPolicy(AutoscalePolicyBean autoscalePolicyBean)
      * @throws RestAPIException
      */
     public static void addApplicationPolicy(ApplicationPolicyBean applicationPolicyBean) throws RestAPIException,
-            AutoscalerServiceInvalidApplicationPolicyExceptionException {
+            AutoscalerServiceInvalidApplicationPolicyExceptionException,AutoscalerServiceApplicationPolicyAlreadyExistsExceptionException {
```
Comment (lahirus) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Same about formatting, add a space before Exception 'AutoscalerServiceApplicationPolicyAlreadyExistsExceptionException'

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java (hunk 1 above)
Comment (pubudu538) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java]: Noted. Thank you!

File: components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java (hunk 2 above)
Comment (imesh) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: +1 for the suggestion, @pubudu538 would you mind sending another PR with proper code formatting?
Thanks

Comment (pubudu538) [inline components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java]: Sent a PR for formatting - https://github.com/apache/stratos/pull/328

**Comment by lahirus** [comment]
Thanks Pubudu.

**Pull request closed without merging**